R/reportCalibration.R

```diff
@@ -0,0 +1,334 @@
+
+  p_renovation <- .readGdxIter(gdx, "p_renovation", maxIter,
+                               asMagpie = FALSE, ttotFilter = tCalib)
+
+  # Read historical gdx files
+  p_stockHist <- .replaceVarName(readGdxSymbol(gdx, "p_stockHist", asMagpie = FALSE))
+
+  p_constructionHist <- .replaceVarName(readGdxSymbol(gdx, "p_constructionHist", asMagpie = FALSE))
+
+  p_renovationHist <- readGdxSymbol(gdx, "p_renovationHist", asMagpie = FALSE)
+
+
+  # COMPUTE DEVIATIONS ---------------------------------------------------------
+
+  p_stockDev <- .computeDeviation(p_stock, p_stockHist)
+
+  p_constructionDev <- .computeDeviation(p_construction, p_constructionHist)
+
+  p_renovationDev <- .computeDeviation(p_renovation, p_renovationHist)
+
+  # Start writing output quantities --------------------------------------------
+
+  out <- list()
+
+  out[["targetFunction"]] <- targetFunction
+
+  out[["stepSize"]] <- stepSize
+
+  # AGGREGATE QUANTITIES -------------------------------------------------------
+
+  # Aggregate across all dimensions
+  out[["stockDevAgg"]] <- .computeSumSq(p_stockDev, rprt = c("iteration", "reg", "typ", "loc", "inc"))
+
+  out[["conDevAgg"]] <- .computeSumSq(p_constructionDev, rprt = c("iteration", "reg", "typ", "loc", "inc"))
+
+  out[["renDevAgg"]] <- .computeSumSq(p_renovationDev, rprt = c("iteration", "reg", "typ", "loc", "inc"))
+
+  out[["flowDevAgg"]] <- .computeFlowSum(out[["conDevAgg"]], out[["renDevAgg"]])
+
+  # Aggregate by heating system (hs)
+  out[["stockDevHs"]] <- .computeSumSq(p_stockDev, rprt = c("iteration", "reg", "typ", "loc", "inc", "hsr"))
+
+  out[["conDevHs"]] <- .computeSumSq(p_constructionDev, rprt = c("iteration", "reg", "typ", "loc", "inc", "hsr"))
+
+  out[["renDevHs"]] <- .computeSumSq(p_renovationDev, rprt = c("iteration", "reg", "typ", "loc", "inc", "hsr"))
+
+  out[["flowDevHs"]] <- .computeFlowSum(out[["conDevHs"]], out[["renDevHs"]])
+
+
+  # COMPUTE RELATIVE AGGREGATE QUANTITIES --------------------------------------
+
+  # Across all dimensions
+  out[["stockDevRel"]] <- .computeRelDev(out[["stockDevAgg"]], p_stockHist, tCalib)
+
+  out[["conDevRel"]] <- .computeRelDev(out[["conDevAgg"]], p_constructionHist, tCalib)
+
+  out[["renDevRel"]] <- .computeRelDev(out[["renDevAgg"]], p_renovationHist, tCalib)
+
+  out[["flowDevRel"]] <- .computeRelDev(out[["flowDevAgg"]], list(p_constructionHist, p_renovationHist), tCalib)
+
+  # Separately for all heating systems (hs)
+  out[["stockDevHsRel"]] <- .computeRelDev(out[["stockDevHs"]], p_stockHist, tCalib, notInHist = "hsr")
+
+  out[["conDevHsRel"]] <- .computeRelDev(out[["conDevHs"]], p_constructionHist, tCalib, notInHist = "hsr")
+
+  out[["renDevHsRel"]] <- .computeRelDev(out[["renDevHs"]], p_renovationHist, tCalib, notInHist = "hsr")
+
+  out[["flowDevHsRel"]] <- .computeRelDev(out[["flowDevHs"]], list(p_constructionHist, p_renovationHist),
+                                          tCalib, notInHist = "hsr")
+
```

Comment:
I modified the computation of the relative deviations by heating system - the idea is that the denominator is computed as the sum of the squares of the historic stock/flow of all heating systems, not just the heating system we are looking at. Therefore, the column ```hsr``` is excluded from the grouping when computing the sum of the squares.